src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs

```diff
@@ -227,9 +236,15 @@ public override StandardValuesCollection GetStandardValues(ITypeDescriptorContex
                 // We need to get the enum values in this rather round-about way so we can filter
                 // out fields marked Browsable(false). Note that if multiple fields have the same value,
                 // the behavior is undefined, since what we return are just enum values, not names.
-                Type reflectType = TypeDescriptor.GetReflectionType(EnumType) ?? EnumType;
+                [UnconditionalSuppressMessage("Trimming", "IL2067:", Justification = "Trimmer does not trim enums")]
+                static Type GetTypeDescriptorReflectionType(Type enumType) => TypeDescriptor.GetReflectionType(enumType);
+
+                Type reflectType = GetTypeDescriptorReflectionType(EnumType) ?? EnumType;
```

Comment:
Ok, this is the kind of thing that would be hidden if we just left this as suppressed on the whole method. Does anything guarantee `TypeDescriptor.GetReflectionType` returns an enum? If I'm reading the code correctly, this is looking at fields of whatever `GetReflectionType` returned and trying to find fields with the same name on the original enum. Looking at the enum is fine, but I don't know how to reason about `GetReflectionType`.

Reply:
This code and comment is a little odd. If we think we already have an Enum type, why do we need to call into TypeDesciptor.GetReflectionType?

It might make sense to go back into the source history for .NET Framework and dig at exactly what scenario this code is enabling.

My concern is that someone is passing in a `Type` into `new EnumConverter` that isn't an Enum, but instead has a `TypeDescriptionProviderAttribute` on it that points to an Enum type. We may be breaking a scenario here with this change.

Reply:
Yep, we feed EnumType (the thing passed to constructor) to a lot of Enum APIs (not just parse, but also Enum.ToObject, Enum.GetUnderlyingType, Enum.IsDefined) so this is not going to work well for non-enum.

However, the type returned from `TypeDescriptor.GetReflectionType` doesn't seem to be fed to enum APIs and only used in this one spot. I don't know how to make sense of it.

Reply:
What I'm ultimately asking for here is to figure out why this code even exists. We already have the enum Type (we think), why are we calling to `TypeDescriptor.GetReflectionType`? Can the `Type` in the constructor not be a RuntimeType? And calling `TypeDescriptor.GetReflectionType` gets the RuntimeType?

Reply:
> I didn't see any explanation in the https://github.com/dotnet/corefx/pull/7454 that added this either.

That change just copied the code from .NET Framework. You can see the exact same comment here: https://referencesource.microsoft.com/#System/compmod/system/componentmodel/EnumConverter.cs,247

Reply:
> `GetReflectionType` is already annotated as requiring and returning PublicParameterlessConstructor | PublicFields so I think we've got that covered.

Didn't realize this part - if that's the case, can we make it so that the result of GetReflectionType doesn't go through the suppressed codepath? I.e. if someone in the future decides to drop the `return:` annotation on this API, we'd get a warning here.

Basically right now `GetPublicStaticEnumFields` assumes the suppression is safe because it's an enum. But the return value from `GetReflectionType` doesn't need a suppression in the first place.